src/main/java/org/cadixdev/mercury/mixin/MixinRemapperVisitor.java

```diff
@@ -244,14 +250,78 @@ else if (rawAnnotation.isSingleMemberAnnotation()) {
                         }
                     }
 
-                    // todo: handle @At remapping
+                    // Remap @At
+                    if (Objects.equals("at", pair.getName().getIdentifier())) {
+                        // it could be a SingleMemberAnnotation here but we don't care about that case
+
+                        if (pair.getValue() instanceof ArrayInitializer) {
+                            // get the annotations in the array
+                            ArrayInitializer arrayInitializer = (ArrayInitializer) pair.getValue();
+                            for (Object expression : arrayInitializer.expressions()) {
+                                if (expression instanceof NormalAnnotation) {
+                                    NormalAnnotation atAnnotation = (NormalAnnotation) expression;
+                                    AtData atDatum = inject.getAtData()[atIndex];
+                                    remapAtAnnotation(ast, declaringClass, atAnnotation, atDatum);
+                                }
+                                atIndex++;
+                            }
+                        }
+
+                        if (pair.getValue() instanceof NormalAnnotation) {
+                            NormalAnnotation atAnnotation = (NormalAnnotation) pair.getValue();
+                            AtData atDatum = inject.getAtData()[atIndex];
+                            remapAtAnnotation(ast, declaringClass, atAnnotation, atDatum);
+                        }
+                    }
                 }
             }
         }
 
         return true;
     }
 
+    private void remapAtAnnotation(AST ast, ITypeBinding declaringClass, NormalAnnotation atAnnotation, AtData atDatum) {
+        for (Object atRaw : atAnnotation.values()) {
+            // this will always be a MemberValuePair
+            MemberValuePair atRawPair = (MemberValuePair) atRaw;
+
+            // check for the target
+            if (Objects.equals("target", atRawPair.getName().getIdentifier())) {
+                // make sure everything is present
+                if (atDatum.getClassName().isPresent()) {
+                    String className = atDatum.getClassName().get();
+                    final StringLiteral originalTarget = (StringLiteral) atRawPair.getValue();
+
+                    // get the class mapping of the class that owns the target we're remapping
+                    final ClassMapping<?, ?> atTargetMappings = this.mappings.computeClassMapping(className).orElse(null);
+                    if (atTargetMappings == null) continue;
+                    atTargetMappings.complete(this.inheritanceProvider, declaringClass);
+                    String deobfTargetClass = atTargetMappings.getFullDeobfuscatedName().replace('.', '/');
```

Comment:
I'll confirm this at a later point, but I'm pretty certain that method should only ever produce names with `/`.